serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/ObjectInspectorUtils.java

```diff
@@ -557,6 +558,12 @@ public static StructField getStandardStructFieldRef(String fieldName,
    */
   public static Field[] getDeclaredNonStaticFields(Class<?> c) {
     Field[] f = c.getDeclaredFields();
+    Arrays.sort(f, new Comparator<Field>() {
+      @Override
+      public int compare(Field a, Field b) {
+        return a.getName().compareTo(b.getName());
+      }
+    });
```

Comment:
This almost seems like a test requirement rather than doing it in production code. Can't we pass the fields in sorted order within the test?